examples/zephyr/common/Kconfig

```diff
@@ -4,6 +4,110 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
+menuconfig GOLIOTH_SAMPLE_COMMON
+	bool "Common helper code shared across multiple samples"
+	help
+	  Build and link common code that is shared across Golioth samples.
+
+if GOLIOTH_SAMPLE_COMMON
+
+rsource "Kconfig.defconfig"
+
+choice GOLIOTH_AUTH_METHOD
+	prompt "Authentication method support"
+
+config GOLIOTH_AUTH_METHOD_PSK
+	bool "PSK-ID / PSK only"
+	imply GOLIOTH_AUTH_PSK_MBEDTLS_DEPS
+	help
+	  Only PSK based authentication will be supported.
+
+config GOLIOTH_AUTH_METHOD_CERT
+	bool "Certificate based only"
+	imply GOLIOTH_AUTH_CERT_MBEDTLS_DEPS
+	help
+	  Only certificate based authentication will be supported.
+
+endchoice
+
+config GOLIOTH_AUTH_PSK_MBEDTLS_DEPS
```

Comment:
I'm not sure it makes sense to have this in the samples directory. Do we need it at all? Did it move from somewhere else?

Reply:
It's currently [in the main Kconfig of the Golioth Zephyr SDK](https://github.com/golioth/golioth-zephyr-sdk/blob/20c97b9cc3f6bd875c26a645734680f78718b96f/net/golioth/Kconfig#L29-L44). We had discussed moving this to samples as the desire was to generally remove PSK support from the CP SDK as only certs should be used in production.

These symbols are [used to populate the client config](https://github.com/golioth/golioth-firmware-sdk/blob/2939c177fc9fa0795e9101acb47d5846b332701d/examples/zephyr/common/hardcoded_credentials.c#L12-L40) in the samples.

Reply:
These `*_DEPS` symbols (and the symbols they select/imply) aren't being used in the code. But authentication was working previously without these options defined, so I'm unclear what it has to do with the examples?

> the desire was to generally remove PSK support from the CP SDK as only certs should be used in production

I think I could have been clearer on this point. We will support PSK and Cert auth in the CP SDK, but we will not support hardcoded credentials (of either type) in the core SDK code (they're only useful for examples and the quickest of prototyping), and we will not be opinionated about how to provision credentials (meaning that things like the settings service won't live in the core SDK).

Reply:
The `*_DEPS` symbols are used in the hello sample. Without it, symbols are missing and communication doesn't function:

```
➜ diff build/zephyr/.config broken.config 
64c64
< CONFIG_GOLIOTH_CIPHERSUITES="TLS_PSK_WITH_AES_128_GCM_SHA256"
---
> CONFIG_GOLIOTH_CIPHERSUITES="TLS_PSK_WITH_AES_128_CBC_SHA256"
112c112
< CONFIG_GOLIOTH_AUTH_PSK_MBEDTLS_DEPS=y
---
> # CONFIG_GOLIOTH_AUTH_PSK_MBEDTLS_DEPS is not set
270c270
< CONFIG_MBEDTLS_KEY_EXCHANGE_PSK_ENABLED=y
---
> # CONFIG_MBEDTLS_KEY_EXCHANGE_PSK_ENABLED is not set
273,274d272
< CONFIG_MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED=y
< CONFIG_MBEDTLS_PSK_MAX_LEN=32
304c302
< CONFIG_MBEDTLS_CIPHER_GCM_ENABLED=y
---
> # CONFIG_MBEDTLS_CIPHER_GCM_ENABLED is not set
```

```
*** Booting Zephyr OS build zephyr-v3.4.0-553-g40d224022608 ***                                                                                                               
[00:00:00.030,000] <wrn> net_sock_tls: No entropy device on the system, TLS communication is insecure!                                                                        
[00:00:00.030,000] <inf> net_config: Initializing network                                                                                                                     
[00:00:00.030,000] <inf> net_config: IPv4 address: 192.0.2.1                                                                                                                  
[00:00:00.030,000] <dbg> hello_zephyr: main: start hello sample                                                                                                               
[00:00:00.030,000] <inf> golioth_samples: Waiting for interface to be up                                                                                                      
[00:00:00.040,000] <inf> golioth_mbox: Mbox created, bufsize: 1100, num_items: 10, item_size: 100                                                                             
[00:00:00.120,000] <inf> golioth_coap_client: Start CoAP session with host: coaps://coap.golioth.io                                                                           
[00:00:00.120,000] <inf> golioth_coap_client: Session PSK-ID: devboard-one-id@ttgo-demo                                                                                       
[00:00:00.120,000] <inf> golioth_coap_client: Entering CoAP I/O loop                                                                                                          
[00:00:02.290,000] <wrn> golioth_coap_client: CoAP message retransmitted                                                                                                      
[00:00:06.620,000] <wrn> golioth_coap_client: CoAP message retransmitted
```

This configuration is what is [used in the Golioth Zephyr SDK](https://github.com/golioth/golioth-zephyr-sdk/blob/bbae6f8419ed52acd74784ed7e876a0f24e75f8b/net/golioth/Kconfig#L32-L78), do you think we should approach it differently?

Reply:
Ah I see, this was getting set in golioth_basics/prj.conf before. I think it would make sense for these options to live in `port/zephyr/Kconfig` then.

Reply:
Okay, these now live in the `port/zephyr/Kconfig` file.